core/kernel/ldelf_loader.c

```diff
@@ -159,27 +161,37 @@ TEE_Result ldelf_init_with_ldelf(struct ts_session *sess,
 	if (res)
 		return res;
 
+	res = bb_memdup_user(arg, sizeof(struct ldelf_arg), &bbuf);
+	if (res)
+		return res;
+
+	arg_bbuf = (struct ldelf_arg *)bbuf;
+
 	if (is_user_ta_ctx(uctx->ts_ctx)) {
 		/*
 		 * This is already checked by the elf loader, but since it runs
 		 * in user mode we're not trusting it entirely.
 		 */
-		if (arg->flags & ~TA_FLAGS_MASK)
+		if (arg_bbuf->flags & ~TA_FLAGS_MASK) {
+			bb_free(bbuf, sizeof(struct ldelf_arg));
```

Comment:
This call isn't needed since the `uctx` is going to be removed anyway.
The rules for the `bb_*()` functions are different from the normal `malloc()` / `free()` pattern since the bounce buffer state is reset for each syscall.

Reply:
Clear explanation, thanks!